### PR TITLE
Cleanup apimachinery tests and stop relying on default conversions

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/kubernetes/issues/87006

The first commit is purely commit, the second is build on top of it, so I recommend reviewing commits separately.

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/runtime/scheme_test.go [hunk 1]
```diff
@@ -978,46 +967,3 @@ func TestMetaValues(t *testing.T) {
 		t.Errorf("Expected %v, got %v", e, a)
 	}
 }
-
-func TestMetaValuesUnregisteredConvert(t *testing.T) {
```
Comment (liggitt) [inline staging/src/k8s.io/apimachinery/pkg/runtime/scheme_test.go:982]: can you summarize why this test was removed? it looks like it's verifying a registered conversion function for an unregistered type still gets called. is that not what we still expect to happen?

File: staging/src/k8s.io/apimachinery/pkg/runtime/testing/conversion.go [hunk 2]
```diff
@@ -0,0 +1,300 @@
+/*
+Copyright 2018 The Kubernetes Authors.
```
Comment (liggitt) [inline staging/src/k8s.io/apimachinery/pkg/runtime/testing/conversion.go:2]: nit: 2020

**Review by liggitt — COMMENTED** [review]
one question, lgtm otherwise

File: staging/src/k8s.io/apimachinery/pkg/runtime/scheme_test.go (hunk 1 above)
Comment (wojtek-t) [reply staging/src/k8s.io/apimachinery/pkg/runtime/scheme_test.go:982]: Good catch. I want to remove it once we stop supporting arbitrary conversions, but it's not now.
Reverted.

**Review by wojtek-t — COMMENTED** [review]
Thanks @liggitt ! PTAL
